### ISSUE TITLE
feat: better doc

### ISSUE THREAD

**Issue by tjbck** [ollama-webui/ollama-webui#432]
Hi guys! I've been noticing some people struggling with installation/update process for ollama-webui and I feel like it would be extremely helpful for a lot of people if we could have a more thorough documentation!

<img width="787" alt="image" src="https://github.com/ollama-webui/ollama-webui/assets/25473318/280c7bbd-f00e-4e3b-8857-1e190908e22c">

Unfortunately, I won't have much time this week and If anyone can get the ball rolling for this (so that I can focus on adding more features), It'll be greatly appreciated. Thanks in advance, guys!

**Comment by justinh-rahb** [comment]
@tjbck I think I can help here, I'll have a look at the docker instructions specifically first.

**Comment by oliverbob** [comment]
I really love this bro @tjbck. And hey, I thank you for all the features in the RAG and full user/admin support that you implemented from my suggestions. You are an amazing, hardworking developer. Will be clicking that sponsor button very, very soon.

I wish we're just one block away from each other so we can better collaborate.

**Comment by oliverbob** [comment]
Why not make me a contributor so I can work on the docs then, work with other devs to work on it too, since your PR is rising like wild fire.

**Comment by svenoaks** [comment]
Idiot here, can someone help me update the docker container? I've got the container with webui only. If I delete the container, then install with the same listed command, it seems I get the same old version (v1.0.0-alpha.40).

I truly tried to search the issues for instructions, but can't find any

**Comment by justinh-rahb** [comment]
> Idiot here, can someone help me update the docker container? I've got the container with webui only. If I delete the container, then install with the same listed command, it seems I get the same old version (v1.0.0-alpha.40).
> 
> I truly tried to search the issues for instructions, but can't find any

Apologies, I am working on an updating doc as I type but it need polish and review. For now, please try:

```bash
docker rm -f ollama-webui
docker pull ollama-webui:main
<your docker run command>
```

I think you probably were missing the docker pull step to get the latest image. Hope this helps you out, and I'll get back to work on documenting it properly 👍

**Comment by svenoaks** [comment]
@justinh-rahb Thank you! it worked, I did have to use `docker pull ghcr.io/ollama-webui/ollama-webui:main` to pull it

**Comment by jukofyork** [comment]
> Idiot here, can someone help me update the docker container? I've got the container with webui only. If I delete the container, then install with the same listed command, it seems I get the same old version (v1.0.0-alpha.40).
> 
> I truly tried to search the issues for instructions, but can't find any

Yeah, I'm pretty tech savy and I struggled badly with this too:

Eventually after stopping, removing and reloading a few times, I thought the rag stuff much be on a different branch so tried changing :main to :rag, etc after seeing those were Github tags.

After none of that worked I eventually went to the URL in the docker command hoping to find the right tag but somehow it went to a Web page that looked like the github and I managed to download something from there that was quite a few 100mb so I knew something was different. 

I've literally no idea how I did it or if I could do it again, and still don't understand why stop+rm of the Docker command didn't seem to work?

**Comment by justinh-rahb** [comment]
Hello @jukofyork I've had to close and reopen the tab I've opened Ollama-WebUI in, and sometimes disable cache to force a complete refresh even after pulling a new image and restarting the container, F5/Cmd+R isn't always enough.

**Comment by tjbck** [comment]
I've been super busy writing a research paper the last couple of days, but I should have more time to work new features for the project starting from tomorrow! @justinh-rahb Thanks a lot for helping, Your efforts haven't gone unnoticed!

**Comment by turnercore** [comment]
This is a docs question related to the new RAG implementation
- What vector store is webui using? 
- Is it doing any sort of long-term storage or memory augmentation? 
- Are the documents being stored as well, or just chunks, or is it just being processed when you add a file?

**Comment by tjbck** [comment]
For everyone's information, One thing I want to emphasize is that people seem to forget that this is a community driven project, which means everyone is dedicating their free time and working on this project alongside their full-time job. So if you guys think the project could use better documentation, I actively encourage to make a PR, be the change you want to see in the world!

**Comment by oliverbob** [comment]
> For everyone's information, One thing I want to emphasize is that people seem to forget that this is a community driven project, which means everyone is dedicating their free time and working on this project alongside their full-time job. So if you guys think the project could use better documentation, I actively encourage to make a PR, be the change you want to see in the world!

Noted @tjbck. Glad to hear.

**Comment by mafrasiabi** [comment]
I have trouble using litellm and create a config file that works with webui, @justinh-rahb  could you please add the config file in docs and add a docker compose file too. I think all of us can benefit from this.

**Comment by 0xb15d3** [comment]
Docker is a life saver for me, especially now that the webui has other backend components. I don't know what the guy above me is talking about.

**Comment by justinh-rahb** [comment]
> After spending an hour trying to get a simple web ui up and running, I'd say get rid of docker, it's a huge stumbling block, to a web interface. It's like having to install an OS in order to run a program. Total overkill.

@iplayfast if you are having trouble setting up Ollama WebUI with Docker, I want to make it clear that switching to a non-Docker setup will not necessarily make the process any easier. The underlying complexity of setting up and configuring a complex web application such as this remains the same, regardless of whether you use Docker or not.

While it may be tempting to look for an easier solution, such as switching to a non-Docker setup, I believe that this will ultimately be more time-consuming and frustrating in the long run.

**Comment by oliverbob** [comment]
@iplayfast  What exactly do yo experience causing you to believe its an overkill? Are you suggesting a branch or fork without the use of docker? What do you have in mind?

**Comment by jukofyork** [comment]
I think one thing that could be improved is an explanation of the steps needed to get the frontend to reflect the changes after an update.

I wrote done all the steps needed to do an update in a text file and even when I followed these again it took a while for the frontend to actually show it had updated and everything (including the version number) stayed the same.

**Comment by stephenhouser** [comment]
I'm putting this in the "better doc" issue, though it may belong elsewhere. I would not rate it as a newbie issue, I've been using Unix/Linux for many years.

This option specification:
` --webui[port=PORT]         Set the port for the web user interface.`

being translated to a command line like:

`./run-compose.sh --webui[port=1234]`

Is completely non-standard. In almost every case brackets `[]` mean an optional argument/parameter and you are not supposed to actually have to type them in. Having to look at the source code to sort it out is not friendly.

**Comment by tjbck** [comment]
@stephenhouser I agree, someone should make a PR to change the behaviour of the `./run-compose.sh` file.

**Comment by jukofyork** [comment]
I still can't seem to get this to update:

```
#!/bin/bash
sudo docker stop ollama-webui
sudo docker rm -f ollama-webui
sudo docker pull ghcr.io/ollama-webui/ollama-webui:main
sudo docker run -d --network=host -v ollama-webui:/app/backend/data -e OLLAMA_API_BASE_URL=http://127.0.0.1:11434/api --name ollama-webui --restart always ghcr.io/ollama-webui/ollama-webui:main
```

(I actually use a different `OLLAMA_API_BASE_URL` because I'm also serving the Ollama API)

```
Ollama Web UI Version
v1.0.0-alpha.100
```

It appears to be doing lots of "stuff" like downloading and unzipping files (unlike when I was just doing `docker run` and it did nothing), but actually getting it to update seems to be completely random - I followed these exact same steps last time to get it to `v1.0.0-alpha.100`...

I should point out I have a degree in CompSci and have been using Unix for 30 years - if I can't get this to update, what hope is there for other people!?

This absolutely needs some completely foolproof documentation on how to update it or else 90% of the likely audience are going to be turned off. IMO the problem isn't using Docker itself; there are plenty of people who happily install/use lots of Linux tools without understanding the step - it's simply that there are no proper instructions...

**Comment by jukofyork** [comment]
Managed to get it to update by looking through the packages and found the name had been changed:
```
#!/bin/bash
sudo docker stop ollama-webui
sudo docker rm -f ollama-webui
sudo docker pull ghcr.io/open-webui/open-webui:main
sudo docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_API_BASE_URL=http://127.0.0.1:11434/api --name open-webui --restart always ghcr.io/open-webui/open-webui:main
```
but you probably don't want to do it this way as it seems to clear the settings and reset the admin login...

**Comment by justinh-rahb** [comment]
https://docs.openwebui.com/migration

**Comment by jukofyork** [comment]
> @jukofyork https://docs.openwebui.com/migration

Thanks - I should have realized the name had changed! :)

**Comment by jukofyork** [comment]
LOL, it's turned back into:

```
Ollama Web UI Version
v1.0.0-alpha.100
```

???

It definitely was "Open Web Ui" earlier today as I noticed the option to turn off the suggested prompts, but for some unknown reason I'm now back to this 😕

**Comment by jukofyork** [comment]
Running the steps in the bash script I posted above has now gone back to:

```
Open WebUI Version
v0.1.104 (latest)
```

but why did it revert itself?

I definitely deleted the old docker image and can confirm `docker ps` shows this:

```
CONTAINER ID   IMAGE                                COMMAND           CREATED         STATUS         PORTS     NAMES
501e6a1dcee5   ghcr.io/open-webui/open-webui:main   "bash start.sh"   6 minutes ago   Up 6 minutes             open-webui
```

**Comment by jukofyork** [comment]
Yeah, I'm beyond confused now:

- There seems to be a copy of the old version running somehow as if I use an ssh tunnel from my phone to this machine I can access the old version and it's fully functional - letting me log in/out, change settings, etc.
- Yet if I use VNC to log into the machine or any other machine on the local network it's showing the new version again (but it magically reverted itself earlier).

I'm beginning to side with the "Docker isn't a good idea" argument now and stuff like this just makes absolutely no sense...

**Comment by tjbck** [comment]
@jukofyork it might just be a browser cache issue :)

**Comment by jukofyork** [comment]
> @jukofyork it might just be a browser cache issue :)

Yeah, but how can it be fully functional and let me use it? Surely if I've deleted the Docker image it should break as soon as I try to use something that needs to write to the backend?

Also earlier today I used the exact same Firefox instance on the same machine to log in and was definitely using the new version, then several hours later the same Firefox instance on the same machine was clearly showing the old version (as was the ssh tunnel)?

**Comment by justinh-rahb** [comment]
> I'm beginning to side with the "Docker isn't a good idea" argument now as stuff like this just makes absolutely no sense...

You don't need Docker to mysteriously end up with multiple dueling installations of Ollama or Open WebUI, we've also seen people manage this without even touching Docker... 🙄

**Comment by jukofyork** [comment]
> > I'm beginning to side with the "Docker isn't a good idea" argument now as stuff like this just makes absolutely no sense...
> 
> You don't need Docker to mysteriously end up with multiple dueling installations of Ollama or Open WebUI, we've also seen people manage to do this without even touching Docker... 🙄

Well I did spend 3 days trying to compile Ollama only to find the service the old version was creating was starting up the old version of the binary it had hidden away somewhere lol.

**Comment by jukofyork** [comment]
Well have cleared both browser caches and they both showing the new version again. Fingers crossed I don't wake up to "Ollama Web UI Version v1.0.0-alpha.100" 😁

**Comment by oliverbob** [comment]
Can you reinstall ollama and the webui from scratch? Lets see how it works.

**Comment by oliverbob** [comment]
Looks like `docker build -t open-webui .` needs to come back in our docs for first time visitors. I'm essentially working with lots of schools these days, and I'm guiding them through ways to install OI. Can someone restore it from the main installation guide? Do you have time for it [justinh-rahb](https://github.com/justinh-rahb) or is it possible?

Thanks.

**Comment by bozo32** [comment]
I volunteer to be an idiot.

I work on OS X (home machine), Linux (without root access, I ssh in, it is a stand alone PC with a titan 12gb gpu) and a HPC that runs scientific linux (some redhat variant) (no docker...uses singularity, have to declare which/how many GPUs I'm using each session, have to close out GPUs after each session to stop billing, it has a bunch of NVIDIA and AMD GPUs. It uses slurm)

Installation on my Mac is trivial

installation on the ssh accessed linux box with cloud flared was a trick
I used a docker setup that I pieced together by reading this docker compose file: https://github.com/jgarland79/ollama-webui-docker
the only problem is that calls from open-webui to ollama containers didn't use ollama/ollama...I had to use the hard internal IP

I have not figured out how to set it up on the cluster in a way that would not be a total pain...every time I initiate a session I have to
1. login
2. pick how many/what kind of gpus I want and get them
3. setup ollama
